lib/VMwareWebService/broker_timeout.rb

```diff
@@ -0,0 +1,64 @@
+module Timeout
```

Comment:
It's not just a question of the thread termination, it also means the timeout exception can be raised outside of the timeout code block. I've seen a number of blog posts attributing various "issues" with timeout (demonstrable by example)  to the critical section in question. All of the described issues had nothing to do with killing y.